src/sentry/static/sentry/app/views/settings/account/accountDetails.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+
+import AsyncView from 'app/views/asyncView';
+import AvatarChooser from 'app/components/avatarChooser';
+import Form from 'app/views/settings/components/forms/form';
+import JsonForm from 'app/views/settings/components/forms/jsonForm';
+import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader';
+import accountDetailsFields from 'app/data/forms/accountDetails';
+import accountPreferencesFields from 'app/data/forms/accountPreferences';
+import {APIRequestMethod} from 'app/api';
+import {updateUser} from 'app/actionCreators/account';
+import {t} from 'app/locale';
+import {User} from 'app/types';
+
+const ENDPOINT = '/users/me/';
+
+class AccountDetails extends AsyncView {
+  getEndpoints(): ReturnType<AsyncView['getEndpoints']> {
+    // local state is NOT updated when the form saves
+    return [['user', ENDPOINT]];
+  }
+
+  handleSubmitSuccess = (user: User) => {
+    // the updateUser method updates our Config Store
+    // No components listen to the ConfigStore, they just access it directly
+    updateUser(user);
+  };
+
+  handleSaveAvatar = (user: User) => {
+    // We need to update the state, because AvatarChooser is using it,
+    // otherwise it will flick
+    this.setState(
+      {
+        user,
+      },
+      () => {
+        this.handleSubmitSuccess(user as User);
+      }
+    );
```

Comment:
In this case, I would restore the previous logic and have everything call `handleSubmitSuccess` (which updates user state), that way we know that `this.state.user` always has the latest `user` value.